### PR TITLE
DEV: Add specs to ChannelSerializer and StructuredChannelSerializer

### PR DESCRIPTION
**Pull request by martin-brennan**
This adds specs to the mentioned serializers to catch regressions
with MessageBus last_ids and to ensure the correct ones are being
returned and passed down to the ChannelSerializer.

Followup to d8ad5c38b9da3315f98c29a2d871aeeeccc85cf8

### CODE REVIEW

File: plugins/chat/app/serializers/chat/structured_channel_serializer.rb [hunk 1]
```diff
@@ -58,11 +58,12 @@ def meta
...
-           chat_message_bus_last_ids[
-             Chat::Publisher.user_tracking_state_message_bus_channel(scope.user.id)
-           ]
-        last_ids[:user_tracking_state] = id
+      if !scope.anonymous? &&
+           user_tracking_state_last_id =
+             chat_message_bus_last_ids[
+               Chat::Publisher.user_tracking_state_message_bus_channel(scope.user.id)
+             ]
+        last_ids[:user_tracking_state] = user_tracking_state_last_id
```
Comment (nattsw) [inline plugins/chat/app/serializers/chat/structured_channel_serializer.rb:66]: Oooh anonymous access? Should this be in the serializer?

Also super personal and I know this is a thing.. but assignments in `if` statements are a lil too much of a side effect for my liking.

File: plugins/chat/spec/serializer/chat/structured_channel_serializer_spec.rb [hunk 2]
```diff
@@ -104,4 +104,103 @@ def fetch_data
...
+            Chat::Publisher.root_message_bus_channel(channel2.id),
+            Chat::Publisher.new_messages_message_bus_channel(channel3.id),
+            Chat::Publisher.new_mentions_message_bus_channel(channel3.id),
+            Chat::Publisher.root_message_bus_channel(channel3.id),
+            Chat::Publisher.new_messages_message_bus_channel(channel4.id),
+            Chat::Publisher.new_mentions_message_bus_channel(channel4.id),
+            Chat::Publisher.root_message_bus_channel(channel4.id),
+          )
+          .returns({})
+        described_class.new(fetch_data, scope: guardian).as_json
```
Comment (nattsw) [inline plugins/chat/spec/serializer/chat/structured_channel_serializer_spec.rb:165]: I would expect `described_class.new(fetch_data, scope: guardian).as_json` to eq that return value above for completion's sake.

Comment (martin-brennan) [reply plugins/chat/spec/serializer/chat/structured_channel_serializer_spec.rb:165]: Yep sure

Comment (martin-brennan) [reply plugins/chat/spec/serializer/chat/structured_channel_serializer_spec.rb:165]: Oh wait that's not correct though -- the point of this spec is to just check that `MessageBus.last_ids` is being called with all the expected channels, and the other tests are similar so the return value doesn't matter too much.

**Review by nattsw — APPROVED** [review]
Thanks!!

**Pull request merged**